### PR TITLE
Add Individual flag

### PR DESCRIPTION
**Pull request by hyanwong**
This allows us to match up with tskit individuals. ~~But the (IMO) more extensive tests currently depend on the default metadata  being `null` (#353 ), so I guess I'll need to figure out how to rework that.~~ (sorry - was being dense, this should work now)

The major change in testing is that I've put the functions to generate complex tree sequences with metadata etc into `tests/tsutil.py`, so that they can be used in both `test_formats.py` and (for round-tripping) in `test_inference.py`. Hope that's OK.

I've also started using the `from_tree_sequence()` function to check that all the necessary data (flags, metadata, etc etc) from a ts is being round-tripped via tsinfer. This seems sensible to me, and we weren't checking it before.

### CODE REVIEW

**Comment by hyanwong** [comment]
Round-trip tests failing, pending a solution to the metadata != `null`  issue. May be worth waiting until we actually sort metadata properly, if we can do that this week (sounds a bit tight, though!)

**Comment by hyanwong** [comment]
Once we [switch to pytest](https://github.com/tskit-dev/tsinfer/issues/351), @benjeffery [points out](https://github.com/tskit-dev/tskit/pull/931#issuecomment-720597295) a nicer way to include a suite of fixed example tree sequences for testing using `conftest.py` with scope="session". I guess we should use that for the functions `get_example_ts`, `get_example_individuals_ts_with_metadata`, and `get_example_historical_sampled_ts` which I have put in tsutil in the current PR.

File: tsinfer/formats.py [hunk 1]
```diff
@@ -1540,6 +1571,7 @@ def add_individual(
         :param list samples_metadata: A list of JSON encodable dict-like objects,
             of length ``ploidy``, giving metadata to be associated with each
             sample, or None to give all samples an empty metadata field.
+        :param list flags: The bitwise flags for this individual.
```
Comment (jeromekelleher) [inline tsinfer/formats.py:1574]: type is int

File: tests/tsutil.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+#
+# Copyright (C) 2018 University of Oxford
```
Comment (jeromekelleher) [inline tests/tsutil.py:2]: 2020?

File: tests/tsutil.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    nodes_individual = tables.nodes.individual  # Assign individuals to sample nodes
+    sample_individuals = np.repeat(
+        np.arange(n, dtype=tables.nodes.individual.dtype), ploidy
+    )
+    if skip_last:
+        # Should work even if some sample nodes are not assigned to an individual
+        sample_individuals[sample_individuals == n - 1] = tskit.NULL
+    nodes_individual[ts.samples()] = sample_individuals
+    tables.nodes.individual = nodes_individual
+    print(tables.populations)
```
Comment (jeromekelleher) [inline tests/tsutil.py:122]: print

**Review by jeromekelleher — COMMENTED** [review]
LGTM, some minor comments.

File: tests/tsutil.py (hunk 2 above)
Comment (hyanwong) [reply tests/tsutil.py:2]: Good catch

File: tests/tsutil.py (hunk 3 above)
Comment (hyanwong) [reply tests/tsutil.py:122]: Damn - I need something to take these stray debugging things out. I always seem to miss one or two. Sorry!

Comment (jeromekelleher) [reply tests/tsutil.py:122]: happens us all

**Comment by hyanwong** [comment]
Done, thanks @jeromekelleher !

**Review by jeromekelleher — APPROVED** [review]

**Comment by jeromekelleher** [comment]
conflict here @hyanwong
